### DISCUSSION TITLE
Comcast killing CableCards (going to Channels/TVE)

### DISCUSSION THREAD

**Discussion by CptKirk-1701** [OpenSageTV/Community#174] [General]
Well Comcast did a firmware update push to CableCards and that killed mine two ago.  They announced that on Oct 24, 2024 that CableCards are now EOL.  New and existing customers cannot get them.  If they die, so sorry, cannot be replaced.  That existing cards would continue to function as long as they had no problems.  Surprise, a firmware push and two perfectly good cards, working for years, both died on the same day and same time!  What a coincidence.

One was in a Tivo, the other in a HDHR.  So all Tivo users have bricks now since they cannot be used on the Comcast system.

So I am looking at Channels and TVE to supplement my OTA.

I found this post on the old forum which looks like it is a good setup guide:

> https://forums.sagetv.com/forums/showthread_t_66804.html?t=66804

But the first thing it states is 

> The following assumes you have opendct installed and configured for generic pipe capture the way ENE outlines at the start. of this thread:
> 
> https://forums.sagetv.com/forums/showthread.php?p=597001&postcount=1

Which of course goes nowhere since it is the old php.  I put it in the HTML format:

> https://forums.sagetv.com/forums/showthread_t_597001.html?t=597001

Buty that just gives a 404 not found error.

Anyone know where that first link is supposed to be going?

Thanks,
Kirk

**Comment by KeithAbb0tt** [comment]
Maybe here: http://forums.sagetv.com/forums/showthread_t_62774.html?t=62774

**Comment by Ugadata** [comment]
Look under   Software -> OpenDCT in the this forum.  There are a few posts about this including links to the Old Sage Forum for setting up Channels, OpenDCT, and an XML EPG Source.

I don't want to discourage you from using TVE, however be aware not all channels are available with TVE.  Support for TVE is starting to wane. 

Due to a recent kerfuffle with moving my Internet from Comcast Business to my Residential cable account, I lost access to all my cable card channels.  I was already frustrated with Comcast/Xfinity so I dropped everything with Comcast/Xfinity.  Maybe I could have gotten it fixed, at least for a short while, but I was aware that cable card support is going away and it was now time to move away from cable.

**Comment by nyplayer** [comment]
Why not just move onto ChannelsDVR ... You can use TVE,Pluto,TUBI TV and Plex live Channels why stick with software that is no longer supported. This is my Setup ... can record from all sources below.... Also they just added the ability to use Channels DVR as a Tuner/source in Plex. XFINITY below is my TVE 147 Channels is not bad at all.

![image](https://github.com/user-attachments/assets/a466a00a-6881-4967-afb7-9625209cdc42)
![image](https://github.com/user-attachments/assets/ef049e8b-789e-4ebe-be6a-5aa305adfa6e)

**Comment by TGsell1** [comment]
Optimum finally pulled my cable card so I was forced to use an AppleTV to stream through a network encoder.    Works great picture wise but I have to manually tune.   How are you tuning your Prime? Xfinity?   Thanks Sent from my iPhoneOn Dec 5, 2024, at 11:51 AM, nyplayer ***@***.***> wrote:﻿
Why not just move onto ChannelsDVR ... You can use TVE,Pluto,TUBI TV and Plex live Channels why stick with software that is no longer supported. This is my Setup
image.png (view on web)

**Reply by nyplayer** [reply]
TVE Guide data and Channels are built in To Channels DVR... You just have to sign in ....

![image](https://github.com/user-attachments/assets/726639f6-895d-415b-afa3-cda2fe30eb7d)


https://getchannels.com/docs/channels-dvr-server/tv-everywhere/availability/#Cablevision

**Comment by LoisGNS** [comment]
I didn't think there was much from Xfinity available on Channels, but it looks like you have it working. My DTA boxes and my other non-X1 box are giving me all sorts of trouble (no doubt Comcast is interfering with them on purpose to force me onto X1). I really really don't want to be forced onto X1 because how do you tune it conveniently? They hype the voice remote, but I'd mostly rather key in channels by number (and have my USBUIRT do so for recordings.) 

Could you say more about your experience with Xfinity & Channels? According to their "check availability" the channels I record most aren't listed.

The only HDHR I have is an ancient one that doesn't tune nearly as well as my Hauppauge cards & dongles, so I use the Hauppauges for my OTA recordings.

**Reply by nyplayer** [reply]
I have X1 and the remotes have numbers .... there is a 30 day trial for Channels try it and post questions there ... this forum sucks in trying to help users.

**Reply by emveepee** [reply]
There is a long thread on this forum on Rogers cable which use Comcast the X1 and  I find the X1 is is quite easy to control with LIRC.  There are no tricks from Comcast you just need to be able to send an Enter after the channel number and Back key to wake up any screensaver.   I am mad I stuck with the old SA boxes for so long.  

I also just got a 4K HDMI grabber (under $100 US) and it works well with it much better than my previous URay and I suspect the BMT series.

**Reply by mickgeller** [reply]
@LoisGNS I have to ask, as I went through a thing recently with Comcast. How old is the wire from the outside pole into your house? Mine was 17 years old and the technician replaced it for me. That fixed my non-X1 connections issues. 

I only have 1 X1 box, and it was working OK. However, my cablecards were giving me fits. After receiving brand new cablecards and having them unable to pair, the drop from the pole outside, causing a loss in signal, was identified as a possible culprit. The tech tested it and his first words to me were "that wire is horrible." He also said that the connection is newer technology and should last a lot longer than the 17 years of the last one.

**Comment by TGsell1** [comment]
Thanks for the update on the inexpensive HDMI grabber that is good news even though I have two urays does anybody out there? Know how to control an Apple TV remotely e.g. on the InternetSent from my iPhoneOn Dec 6, 2024, at 4:10 PM, emveepee ***@***.***> wrote:﻿
There is a long thread on this forum on Rogers cable which use Comcast the X1 and  I find the X1 is is quite easy to control with LIRC.  There are no tricks from Comcast you just need to be able to send an Enter after the channel number and Back key to wake up any screensaver.   I am mad I stuck with the old SA boxes for so long.
I also just got a 4K HDMI grabber (under $100 US) and it works well with it much better than my previous URay and I suspect the BMT series.

**Comment by LoisGNS** [comment]
Thanks for the X1 info; I'll have to see if I can find documentation on it. Being able to just enter channel numbers would make it a much more acceptable setup than being forced to use voice or wade through a lot of menus just to watch channel 59 or whatever. 

Of course, it will be whatever they replace my Comcast DTAs with that I'll need to find a way to control with my USBUIRT, but if they will accept IR-blaster output & the USBUIRT can learn the remote codes (or I can find them ready-documented somewhere), that could work.

I looked up LIRC, and didn't see how it would work for me. It seems to be Linux or MacOS only, though there may be a Windows port. But even then, what kind of hardware would I need? I didn't see USBUIRT on the supported list, but maybe since I have a USBUIRT I wouldn't need LIRC? I'd be satisfied if whatever secondary boxes come with X1 can be tuned via USBUIRT and output a signal that I can use with my Hauppauge tuners or satisfactory replacements. My current DTAs output analog channel 3/4, which is supported by my Hauppauge cards, although those may also accept composite inputs instead (but not HDMI absent some sort of converter to something that they do accept). 

What do you connect the 4K HDMI grabber to? I don't have or need 4K; in fact, I'm currently just using the standard-def channel 3/4 output for my cable shows - which is adequate for me.

**Reply by emveepee** [reply]
WINLIRC works too but it is device specific.  Right now because my HDMI grabber is near a TV and away from my server, I use an old RPi for LIRC tuning.  The device I got is discussed here https://forums.nextpvr.com/showthread.php?tid=65506&pid=596348#pid596348 and it connects to the HDMI out from the XI.  On  the same grabber I also record via a USB UVB device and the grabber does the MPEG2 conversion.  This relieves the server from doing any work.  

I posted a link to a video of the grabber tuning and you can see it is quite fast (you can see the XI GUI)  I mainly posted that to show the HEVC tuning (first tune to 507) because it is little slower than H264 (I need to add a timeout in my tuning script to hide that green screen or just stick with h264.

**Comment by LoisGNS** [comment]
@mickgeller. The wire is quite old - at least I don't remember when it was last replaced. However, when we had the tech out in August, they claimed that they didn't need to check or replace it because they were able to see "from their end" that the signal strength was good;  and that the only solution was to replace our boxes because somehow they all talk to each other (even though the little ones I use with Sage aren't interactive). I suspect that was BS and they just wanted to push us into X1. So the trick is, how do we get them to _really_ check/replace the connecting wire?

To top things off I just got yet another notice of a price increase.

**Reply by mickgeller** [reply]
The tech came out to my house and tested the line at the point the wire entered the house. How to get them to do that, when they don't want to, is a mystery to me. For me, coming out to my house is something that they were suggesting from the very beginning when I thought it was a cablecard issue. I was thinking I did not need them at my house to remedy bad cablecards. I was wrong. 

There are Comcast user forums on their web site. Scanning and searching there my prove fruitful to learn how to get them to respond. You can find it by searching for "xfinity user forum."

**Reply by TGsell1** [reply]
My cable card was discontinued of course so I decided a workaround but the tuning is manual.  Basically I use the Google Home app on a Fire tablet to control the android device my cable company provided.   Using Vysor I can mirror control the tablet/remote on a windows computer which I can thus control remotely using VNC or RD.   My question; can anyone think of a way to tune the app via Sage? Can I send code via the Windows computer to the tablet which will control the Google 'Home'app?

**Reply by Zogg44** [reply]
Sage has plugins to run a script for tuning, Gentuner for Linux and EXEMultiTuner for Windows. I've used them both, and they are similar but different because of the way the OSes work. If you are running Sage on Windows then it will actually be easier. There are several ways to approach this depending on your setup.

1. Let Sage control a USB-UIRT to change the channel. This is probably the easiest way, assuming the Android box has a remote control.
2. Run Android adb (Windows only) to send commands to the Android app. This might not work depending on if/how the app receives commands over adb. Will take some experimentation on your part, but someone may have already done this. I'd do some searching to see if there are already solutions or examples of this. 
3. Leave the VNC/RD connection up all the time, and use a script to move the mouse and/or enter key presses to set the channel. This is very Rube Goldberg-esque (i.e. the old Mousetrap game) but could work. I have automated this type of process before using AutoIt, but you could also use NirCmd commands in a shell script to do the same thing.

As an example, in the past running Sage in a Docker, I have used Gentuner to run a script that used Curl to sent the tuner and channel in a URL to a Windows machine running EventGhost, that then stripped off the channel number and sent it to an AutoIt script that opened a web browser to Spectrum TV and set the proper channel. Crazy, I know, but sending commands between Linux and Windows is problematic, and I know EventGhost very well. Oh, and it worked great. I do something similar now but from ChannelsDVR running in a docker to the Windows machine. 

It's a bit easier if it's all on one Windows machine, as you can have EXEMultiTuner call the script that would send the channel over adb or do keystroke/mouse commands to control the VNC window.

Sorry if this is all over the map. I've tried to give you a few options to start with. I no longer use Sage but I do still have all the files and some write-ups.

**Comment by TGsell1** [comment]
This is very helpful!  I have used the exe multi tuner before for windows media center which is now useless.  Usbuirt probably NG since the Android remote is Bluetooth.  Anything thoughts on a Bluetooth remote code akin to the usbuirt?  Or can the exe multi be scripted to tune the Android box ?  The rub Goldberg approach is not terrible since I have a dedicated windows machine to devote to this.  Thanks !Sent from my iPhoneOn Jan 2, 2025, at 6:24 PM, Zogg ***@***.***> wrote:﻿
Sage has plugins to run a script for tuning, Gentuner for Linux and EXEMultiTuner for Windows. I've used them both, and they are similar but different because of the way the OSes work. If you are running Sage on Windows then it will actually be easier. There are several ways to approach this depending on your setup.

Let Sage control a USB-UIRT to change the channel. This is probably the easiest way, assuming the Android box has a remote control.
Run Android adb (Windows only) to send commands to the Android app. This might not work depending on if/how the app receives commands over adb. Will take some experimentation on your part, but someone may have already done this. I'd do some searching to see if there are already solutions or examples of this.
Leave the VNC/RD connection up all the time, and use a script to move the mouse and/or enter key presses to set the channel. This is very Rube Goldberg-esque (i.e. the old Mousetrap game) but could work. I have automated this type of process before using AutoIt, but you could also use NirCmd commands in a shell script to do the same thing.

As an example, in the past running Sage in a Docker, I have used Gentuner to run a script that used Curl to sent the tuner and channel in a URL to a Windows machine running EventGhost, that then stripped off the channel number and sent it to an AutoIt script that opened a web browser to Spectrum TV and set the proper channel. Crazy, I know, but sending commands between Linux and Windows is problematic, and I know EventGhost very well. Oh, and it worked great. I do something similar now but from ChannelsDVR running in a docker to the Windows machine.
It's a bit easier if it's all on one Windows machine, as you can have EXEMultiTuner call the script that would send the channel over adb or do keystroke/mouse commands to control the VNC window.
Sorry if this is all over the map. I've tried to give you a few options to start with. I no longer use Sage but I do still have all the files and some write-ups.

**Reply by Zogg44** [reply]
I'm not sure about bluetooth. If you also run Home Assistant, it works with bluetooth devices and might detect it. I'm not sure if you could control it though, or if there's even an integration for HA to recognize the android box.

I would take a close look at the box and look for an IR window. Also check the manual and online to see if anyone mentions using an IR remote to control it.

Barring that, I'd install the android SDK to get adb, then try interrogating the device to see what commands it lists. Also, search online to see what others have done with it. If someone has figured out some command structure to control it, then it can be scripted 
to work with exemultituner.

**Comment by TGsell1** [comment]
Great advice. Thanks Sent from my iPhoneOn Jan 2, 2025, at 9:47 PM, Zogg ***@***.***> wrote:﻿
I'm not sure about bluetooth. If you also run Home Assistant, it works with bluetooth devices and might detect it. I'm not sure if you could control it though, or if there's even an integration for HA to recognize the android box.
I would take a close look at the box and look for an IR window. Also check the manual and online to see if anyone mentions using an IR remote to control it.
Barring that, I'd install the android SDK to get adb, then try interrogating the device to see what commands it lists. Also, search online to see what others have done with it. If someone has figured out some command structure to control it, then it can be scripted
to work with exemultituner.

**Comment by TGsell1** [comment]
Also I can access encrypted tv via an Apple TV but haven’t figured out how to tune it other than manually.  Sent from my iPhoneOn Jan 2, 2025, at 9:47 PM, Zogg ***@***.***> wrote:﻿
I'm not sure about bluetooth. If you also run Home Assistant, it works with bluetooth devices and might detect it. I'm not sure if you could control it though, or if there's even an integration for HA to recognize the android box.
I would take a close look at the box and look for an IR window. Also check the manual and online to see if anyone mentions using an IR remote to control it.
Barring that, I'd install the android SDK to get adb, then try interrogating the device to see what commands it lists. Also, search online to see what others have done with it. If someone has figured out some command structure to control it, then it can be scripted
to work with exemultituner.

**Reply by TGsell1** [reply]
I think I am close to getting this working -- I have a dedicated windows PC with Google Home - connected to the Optimum Stream as a 'remote' ,, Its full screen and all you have to do on the PC is type in a number to tune the Optimum Stream.   So how do I connect the EXEMultituner signal from Sage to send a number (as it was sending to my Windows Media Center) to this PC ?  I forget what I did to get the WMC to connect to EXE Multi... was it through an IP address?  .....

**Reply by Zogg44** [reply]
> I think I am close to getting this working -- I have a dedicated windows PC with Google Home - connected to the Optimum Stream as a 'remote' ,, Its full screen and all you have to do on the PC is type in a number to tune the Optimum Stream. So how do I connect the EXEMultituner signal from Sage to send a number (as it was sending to my Windows Media Center) to this PC ? I forget what I did to get the WMC to connect to EXE Multi... was it through an IP address? .....

I've attached a very old guide on setting up EXE Multi Tuner, but it will give you the gist of things. Assuming you have EMT installed, you need a registry setting to point to the batch file. Here's an example of one I used. 

```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Frey Technologies\Common\EXEMultiTunerPlugin]
"command"="c:\\scripts\\exemultitunerscript.bat %DEVICE% %CHANNEL%"
```

Then, the batch file will get that info like this:

```
@echo off
echo %date% %time% device is %1 channel is %2 >c:\scripts\exemultitunerinfo.txt
```

The tricky part is that %2 will be a text string, but you need to translate that to key presses. NirCmd (nirsoft.net/utils/nircmd.html) can send key presses but I don't think it can translate for example "084" into three keypresses of "0", "8", and "4". I use a python script in EventGhost to do this, so you might have to do something like have the exemultituner.bat script then call a python or perl script to break out the digits.

For now, you can download NirCmd and play with it to test sending keystrokes. It might be possible to break out the characters in a batch file, I haven't looked yet. Or have the batch file call a python script to break them out and send them as keystrokes. But see the note below about the window focus.

Another method is using AutoIt. You can compile an AutoIt script into an exe also, which is nice. Check out the example script at the bottom of this page https://www.autoitscript.com/autoit3/docs/functions/Send.htm, you can see it sending a text message into Notepad. Instead, you could send "084" to the app.

Also note that it is best to make sure that the desired window has focus, as Windows has a bad habit of moving the focus to other apps/processes. AutoIt has a tool you can find the apps handle and then you can use a command to make sure it has the focus.


[STB Firewire Channel Changing with ExeMultiTuner (rev 1.02).pdf](https://github.com/user-attachments/files/18303622/STB.Firewire.Channel.Changing.with.ExeMultiTuner.rev.1.02.pdf)

**Comment by TGsell1** [comment]
Thanks for the info!  I forgot about the registry entry and will try the script help you gave.  Sent from my iPhoneOn Jan 3, 2025, at 3:53 PM, Zogg ***@***.***> wrote:﻿

I think I am close to getting this working -- I have a dedicated windows PC with Google Home - connected to the Optimum Stream as a 'remote' ,, Its full screen and all you have to do on the PC is type in a number to tune the Optimum Stream. So how do I connect the EXEMultituner signal from Sage to send a number (as it was sending to my Windows Media Center) to this PC ? I forget what I did to get the WMC to connect to EXE Multi... was it through an IP address? .....

I've attached a very old guide on setting up EXE Multi Tuner, but it will give you the gist of things. Assuming you have EMT installed, you need a registry setting to point to the batch file. Here's an example of one I used.
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Frey Technologies\Common\EXEMultiTunerPlugin]
"command"="c:\\scripts\\exemultitunerscript.bat %DEVICE% %CHANNEL%"

Then, the batch file will get that info like this:
@echo off
echo %date% %time% device is %1 channel is %2 >c:\scripts\exemultitunerinfo.txt

The tricky part is that %2 will be a text string, but you need to translate that to key presses. NirCmd (nirsoft.net/utils/nircmd.html) can send key presses but I don't think it can translate for example "084" into three keypresses of "0", "8", and "4". I use a python script in EventGhost to do this, so you might have to do something like have the exemultituner.bat script then call a python or perl script to break out the digits.
For now, you can download NirCmd and play with it to test sending keystrokes. It might be possible to break out the characters in a batch file, I haven't looked yet. Or have the batch file call a python script to break them out and send them as keystrokes. But see the note below about the window focus.
Another method is using AutoIt. You can compile an AutoIt script into an exe also, which is nice. Check out the example script at the bottom of this page https://www.autoitscript.com/autoit3/docs/functions/Send.htm, you can see it sending a text message into Notepad. Instead, you could send "084" to the app.
Also note that it is best to make sure that the desired window has focus, as Windows has a bad habit of moving the focus to other apps/processes. AutoIt has a tool you can find the apps handle and then you can use a command to make sure it has the focus.
STB Firewire Channel Changing with ExeMultiTuner (rev 1.02).pdf
